packages/workshop-app/app/utils/mdx.tsx

```diff
@@ -6,6 +6,46 @@ import { LaunchEditor } from '~/routes/launch-editor'
 import { AnchorOrLink } from './misc'
 import Accordion from '~/components/accordion'
 
+function notification(button: EventTarget & HTMLButtonElement, on?: boolean) {
+	if (button) {
+		const label = button.previousElementSibling
+		if (on) label?.removeAttribute('hidden')
+		else label?.setAttribute('hidden', 'true')
+		button.style.backgroundColor = on ? '#e5e5e5' /* bg-gray-200 */ : ''
+	}
+}
+
+export function PreWithCopyToClipboard({ children, ...props }: any) {
+	const showCopyButton = !Object.keys(props).find(att => att === 'data-nocopy')
+
+	return (
+		<div className="group relative">
+			{showCopyButton ? (
+				<div className="absolute top-0 right-0 z-50 m-2 mr-2 flex items-center gap-2 opacity-0 transition duration-300 ease-in-out group-hover:opacity-100">
```

Comment:
So it won't focus with opacity of 0? What if we do `opacity-[0.01]`?

Reply:
`focus-within:opacity-100` did it